### PR TITLE
Check size class instead of device type

### PR DESCRIPTION
**Pull request by enhorn**
95% of the changes is in the example demo project, where I updated it to do any size class checks in runtime instead of launch time. 😅

### CODE REVIEW

File: Form/SeparatorStyle.swift [hunk 1]
```diff
@@ -69,11 +69,22 @@ public extension UITraitCollection {
...
+    var isRegular: Bool { return horizontalSizeClass == .regular }
+
+    /// Returns true if horizontal size class is `.compact`
+    var isCompact: Bool { return horizontalSizeClass == .compact }
+}
+
+// Deprecated
+public extension UITraitCollection {
+
+    @available(*, deprecated, message: "use `isRegular` instead")
```
Comment (nataliq) [inline Form/SeparatorStyle.swift]: Since `isRegular` and `userInterfaceIdiom == .pad` are not equivalent and there will definitely be cases in which the use of `userInterfaceIdiom` is valid, perhaps the message can say something like: 
"No longer supported as a convenience variable. Prefer size class checks (isRegular/isCompact) or check the `userInterfaceIdiom` manually".

File: Form/SectionStyle.swift [hunk 2]
```diff
@@ -183,9 +183,9 @@ extension SectionStyle.Background {
 }
 
 private extension TextStyle {
-    static let headerFooterTextStyle = TextStyle(font: .systemFont(ofSize: 13), color: UIColor(red: 0.43, green: 0.43, blue: 0.45, alpha: 1), alignment: .left)
-    static let headerTextStyle = headerFooterTextStyle.uppercased
-    static let footerTextStyle = headerFooterTextStyle
+    static var headerFooterTextStyle: TextStyle { return TextStyle(font: .systemFont(ofSize: 13), color: UIColor(red: 0.43, green: 0.43, blue: 0.45, alpha: 1), alignment: .left) }
```
Comment (nataliq) [inline Form/SectionStyle.swift]: This change doesn't seem needed, am I missing something?

File: Form/SeparatorStyle.swift (hunk 1 above)
Comment (enhorn) [reply Form/SeparatorStyle.swift]: True

File: Form/SectionStyle.swift [hunk 3]
```diff
@@ -60,8 +60,8 @@ public struct DynamicSectionStyle: DynamicStyle {
 }
 
 public extension DynamicSectionStyle {
-    static let systemGrouped = DynamicSectionStyle { Style(traits: $0, isGrouped: true) }
-    static let systemPlain = DynamicSectionStyle { Style(traits: $0, isGrouped: false) }
+    static var systemGrouped: DynamicSectionStyle { return DynamicSectionStyle { Style(traits: $0, isGrouped: true) } }
```
Comment (nataliq) [inline Form/SectionStyle.swift]: Why is this change needed, isn't DynamicSectionStyle taking a block with trait collection and can be evaluated for different ones?
